### PR TITLE
Updated to Facebook SDK 4.1

### PR DESCRIPTION
**Pull request by puf**
@jdimond Can you review?

We now have Facebook SDK as a dependency, instead of including it in
the project.
Facebook SDK 4 removed the Session object, so reworked the code for
LoginButton and AccessTokenTracker

### CODE REVIEW

File: app/src/main/java/com/firebase/samples/logindemo/MainActivity.java [hunk 1]
```diff
@@ -123,14 +130,16 @@ protected void onCreate(Bundle savedInstanceState) {
         /* *************************************
          *              FACEBOOK               *
          ***************************************/
-        /* Load the Facebook login button and set up the session callback */
+        /* Load the Facebook login button and set up the tracker to monitor access token changes */
+        mFacebookCallbackManager = CallbackManager.Factory.create();
         mFacebookLoginButton = (LoginButton) findViewById(R.id.login_with_facebook);
-        mFacebookLoginButton.setSessionStatusCallback(new Session.StatusCallback() {
+        mFacebookAccessTokenTracker = new AccessTokenTracker() {
```
Comment (jdimond) [inline app/src/main/java/com/firebase/samples/logindemo/MainActivity.java]: You never call startTracking. Does creating the class automatically do that?

Comment (puf) [inline app/src/main/java/com/firebase/samples/logindemo/MainActivity.java]: Wow, you're really paying attention. I got most of this from the Scrumptious sample in the Facebook SDK, which -guess what- also doesn't call startTracking. Given that tracking works, it indeed seems to track automatically. The Facebook docs are also quite terse on its intended use: https://developers.facebook.com/docs/reference/android/current/class/AccessTokenTracker/ "This class can be extended to receive notifications of access token changes. The AccessTokenTracker.stopTracking() method should be called in the onDestroy() method of the receiving Activity or Fragment."

Comment (jdimond) [inline app/src/main/java/com/firebase/samples/logindemo/MainActivity.java]: Seems like it. isTracking() returns true...

**Comment by jdimond** [comment]
lgtm. Thanks

**Pull request merged**